dev/com.ibm.ws.jca_fat_bval/fat/src/com/ibm/ws/jca/fat/bval/BeanValidationTest.java

```diff
@@ -0,0 +1,201 @@
+/*******************************************************************************
+ * Copyright (c) 2014, 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jca.fat.bval;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+@RunWith(FATRunner.class)
+public class BeanValidationTest extends FATServletClient {
+    private static final String BVAL_APP = "jca-bval";
+    private static final String BVAL_RAR = "BvalRA";
+
+    @Server("com.ibm.ws.jca.fat.bval")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create applications
+        ShrinkHelper.defaultApp(server, BVAL_APP, "web", "web.mdb");
+        ShrinkHelper.defaultRar(server, BVAL_RAR, "com.ibm.bval.jca.adapter");
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        if (server.isStarted())
+            server.stopServer();
+    }
+
+    @After
+    public void afterEach() throws Exception {
+        if (server.isStarted())
+            server.stopServer();
+    }
+
+    /**
+     * Utility method to run a test on RABValServlet.
+     *
+     * @param query query string for the servlet
+     * @return output of the servlet
+     * @throws IOException if an error occurs
+     */
+    private StringBuilder runInServlet(String queryString) throws Exception {
+        return runTestWithResponse(server, BVAL_APP, "testBeanValidation&" + queryString);
+    }
+
+    @Test
+    public void testAddAndFind() throws Exception {
```

Comment:
can we evaluate the runtime of these test method? Since this one is doing a server restart I expect it takes >5s -- can we move this test and other long-running tests to FULL mode?